### PR TITLE
Add support for github generated release notes, closes #51

### LINKED ISSUE #51: Add support for github generated changelogs
Github allows for automatically generated changelogs on a release: https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#configuring-automatically-generated-release-notes

This is available through their [CLI `gh` tool](https://cli.github.com/manual/gh_release_create) as `--generate-release-notes`. Looking at this repo a bit it seems like it uses the API so that has an option too: https://docs.github.com/en/rest/releases/releases#generate-release-notes-content-for-a-release

It would be awesome if this library had something similar, something maybe like 

```kotlin
githubRelease {
    repo("repo")
    tagName(semver.versionTagName)
    body(autoGenerated())               // <-- something like this
}
```

I might be able to help with this contribution depending on if it's something that would be possible to do 👍

### PR DESCRIPTION
**Pull request by ahatzz11**
Fixes #51, which is around being able to use the generated release notes that github provides.

This is natively supported in the github API to create a release, so I just piggybacked onto that API call: https://docs.github.com/en/rest/releases/releases#create-a-release

I set the default to `false` to match what github does, and to prevent duplicate logs in situations like using `body(changelog())`. If it is set to `true` and a body is also provided, the body will be pre-pended to the generated notes. If a name is not provided, one will be generated (I believe it uses the tag based on my limited testing).

Kotlin usage:
```kotlin
githubRelease {
    generateReleaseNotes(true)
}
```

Groovy usage:
```groovy
githubRelease {
    generateReleaseNotes.set true
}
```

Each teams is also able to customize the syntax and groupings of the release that is generated by github with a `.github/release.yaml` file in the repo (with no work on the plugin side). More information here: https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#configuring-automatically-generated-release-notes (<- This kind of extra info might be helpful to have for people in the wiki).

---

I'm a first time contributor, so let me know if there's anything I missed!
I'm also happy to update the wiki with this new parameter, but it doesn't look like I have access to it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +35,9 @@ This plugin uses [OkHttp](http://square.github.io/okhttp/) to send a POST reques
 If you are using multiple GithubRelease tasks to compose a single release, the release must be published `draft=false` with the first task. Currently, the plugin cannot find an existing release if it is a draft.
 
 ## Changelog
+2.4.0
```
Comment (ahatzz11) [inline README.md:38]: 📝 I'm not sure how strictly this repo follows semver, but as a new feature I thought a minor bump would be appropriate.

**Comment by BreadMoirai** [comment]
Looks good to me! Thanks for the contribution

**Review by BreadMoirai — APPROVED** [review]

**Comment by ahatzz11** [comment]
@BreadMoirai how do the releases happen? I'd love to start using this.

**Comment by BreadMoirai** [comment]
I've published the release as 2.4.1
I've also opened the wiki for anyone to edit if you wanted to update that as well.

**Comment by BreadMoirai** [comment]
Since the plugin is published to the gradle plugin portal, only I can publish it as it's under my account currently
If you wanted to use your dev branch, you can run the `publishToMavenLocal` task and consume it from there. See https://docs.gradle.org/current/userguide/declaring_repositories.html#sub:maven_local

**Comment by ahatzz11** [comment]
Awesome thanks! I just updated the wiki as well: https://github.com/BreadMoirai/github-release-gradle-plugin/wiki#generatereleasenotes

I did so some small testing locally with `publishToMavelLocal` - thanks again!

**Pull request merged**